tx-pool/src/process.rs

```diff
@@ -1003,27 +988,23 @@ fn _submit_entry(
     let tx_hash = entry.transaction().hash();
     match status {
         TxStatus::Fresh => {
-            if tx_pool.add_pending(entry.clone()) {
-                debug!("submit_entry pending {}", tx_hash);
+            if tx_pool.add_pending(entry.clone()).unwrap_or(false) {
```

Comment:
If `tx_pool.add_pending()` is not `Ok`, the reason may not be `Reject::Duplicated`. It may be `Reject::ExceededMaximumAncestorsCount` throw by 
 `fn add_entry` => `self.record_entry_links`.